bundles/org.openhab.binding.bsblan/src/main/java/org/openhab/binding/bsblan/internal/handler/BsbLanBaseThingHandler.java

```diff
@@ -0,0 +1,128 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.bsblan.internal.handler;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+
+import org.openhab.binding.bsblan.internal.configuration.BsbLanBridgeConfiguration;
+import org.openhab.binding.bsblan.internal.api.BsbLanApiCaller;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Basic Handler class for all BSB-LAN services.
+ *
+ * @author Peter Schraffl - Initial contribution
+ */
+@NonNullByDefault
+public abstract class BsbLanBaseThingHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(BsbLanBaseThingHandler.class);
+    private @Nullable BsbLanBridgeHandler bridgeHandler;
+
+    public BsbLanBaseThingHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            updateChannel(channelUID.getId());
+        } else {
+            setChannel(channelUID.getId(), command);
+        }
+    }
+
+    @Override
+    public void initialize() {
+        BsbLanBridgeHandler bridgeHandler = getBridgeHandler();
+        if (bridgeHandler == null) {
+            logger.debug("Initializing '{}': thing is only supported within a bridge", getDescription());
+            updateStatus(ThingStatus.OFFLINE);
+            return;
+        }
+        logger.debug("Initializing '{}' thing", getDescription());
+        bridgeHandler.registerThing(this);
+    }
+
+    protected synchronized @Nullable BsbLanBridgeHandler getBridgeHandler() {
```

Comment:
Why is this synchronized?

Reply:
To be honest, this was initially copied from the fronius binding and I thought it might be required due to the openhabs threading model and refreshing things in combination with the usage of BaseThingHandlers `scheduler`. Haven't therefore had a deeper look.
But for what I see (without knowing the internal openHab logic) there shouldn't be an issue - except that `bridgeHandler` member might be retrieved multiple times?

Reply:
It should be safe to remove synchronization here